src/Microsoft.Azure.SignalR.Management/ServiceManager.cs

```diff
@@ -28,8 +28,10 @@ internal class ServiceManager : IServiceManager
         private readonly string _productInfo;
         private readonly ServiceManagerContext _context;
         private readonly RestClientFactory _restClientFactory;
+        private readonly ServiceProvider _serviceProvider;
 
-        internal ServiceManager(ServiceManagerContext context, RestClientFactory restClientFactory)
+        //in builder mode, serviceProvider is not null and should be disposed.
+        internal ServiceManager(ServiceManagerContext context, RestClientFactory restClientFactory, ServiceProvider serviceProvider = null)
```

Comment:
when would `ServiceProvider` be null?

Reply:
Yes, some mistakes here . Changed it to `IServiceProvider` from `ServiceProvider` and make it required .